skill-tree /src/test/java/com/RDS/skilltree/SkillsIntegrationTests.java

```diff
@@ -1,137 +1,250 @@
-                .body("data[0].description",isA(String.class));
+                .body("content", hasSize(0))
+                .body("totalPages", equalTo(0))
+                .body("totalElements", equalTo(0))
+                .body("last", equalTo(true))
+                .body("size", equalTo(100))
+                .body("number", equalTo(0))
+                .body("numberOfElements", equalTo(0))
+                .body("empty", equalTo(true));
     }
 
     @Test
-    @Disabled
-    @DisplayName("Return all skills given userId")
-    public void testAPIReturnsSkillsGivenUser() {
+    @DisplayName("Return 200, on skill found given skillId")
+    public void testAPIReturns200_OnSkillFoundById() {
+        UUID skillId = skill.getId();
+
         Response response = given()
-                .get("/skills/u1");
+                .pathParam("id", skillId)
+                .get("/v1/skills/{id}");
 
         response.then()
                 .statusCode(200)
                 .contentType("application/json")
-                .body("code",equalTo(200))
-                .body("data", hasSize(1))
-                .body("data[0].id", equalTo("s1"))
-                .body("data[0].name",equalTo("Java"))
-                .body("data[0].type", anyOf(equalTo("ATOMIC"), equalTo("DERIVED")))
-                .body("data[0].description",isA(String.class));
+                .body("id", equalTo(String.valueOf(skillId)))
+                .body("name", equalTo("Java"))
+                .body("type", equalTo("ATOMIC"));
     }
 
     @Test
-    @Disabled
-    @DisplayName("Return 400 on incorrect userId")
-    public void testAPIReturns400_OnIncorrectUserId() {
+    @DisplayName("Return 404, on skill not found given SkillId")
+    public void testAPIReturns404_OnSkillNotFound() {
+        UUID skillId = UUID.randomUUID();
         Response response = given()
-                .get("/skills/{userId}");
+                .pathParam("id", skillId)
+                .get("/v1/skills/{id}");
 
         response.then()
-                .statusCode(400)
-                .body("code",equalTo(400))
-                .body("message",equalTo("Invalid userId passed"));
+                .statusCode(404)
+                .body("message", equalTo("Skill not found with given Id"));
 
     }
 
     @Test
-    @Disabled
-    @DisplayName("Return 404 on, no skills found on given userId")
-    public void testAPIReturns404_OnNoSkillsFound() {
+    @DisplayName("Return 200, on skill with given name")
+    public void testAPIReturns200_OnSkillFoundGivenName() {
         Response response = given()
-                .get("/skills/{userId}");
+                .pathParam("name", "Java")
+                .get("/v1/skills/name/{name}");
+
+        response.then()
+                .statusCode(200)
+                .contentType("application/json")
+                .body("name", equalTo("Java"))
+                .body("type", equalTo("ATOMIC"));
```

Comment:
Can we also assert on `users` and `id`

Reply:
have checked on Id of the skill, for the user we will have to check in user integration tests as we add a skill to the user. So it will be possible in User integration tests.